tools/flaky-test-reporter/jsonreport/jsonreport.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jsonreport
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"sort"
+	"time"
+
+	"github.com/knative/test-infra/shared/common"
+	"github.com/knative/test-infra/shared/prow"
+)
+
+const (
+	filename = "flaky-tests.json"
+	jobName  = "ci-knative-flakes-reporter" // flaky-test-reporter's Prow job name
+	maxAge   = 4                            // maximum age in days that JSON data is valid
+)
+
+// Report contains concise information about current flaky tests
+type Report struct {
+	Repo  string   `json:"-"`
+	Flaky []string `json:"flaky"`
+}
+
+func (r *Report) writeToArtifactsDir() error {
+	err := common.CreateDir(path.Join(prow.GetLocalArtifactsDir(), r.Repo))
```

Comment:
you need the artifactsDir var as you are using it on L48. 

I meant combine the if condition below and this line

Reply:
sorry, the line preview window confused me. It should be good to go now